urlencoder-app/build.gradle.kts

```diff
@@ -70,28 +78,22 @@ tasks {
         dependsOn(fatJar)
     }
 
-    withType<GenerateMavenPom>().configureEach {
-        destination = file("$projectDir/pom.xml")
-    }
```

Comment:
I removed this because now each subproject has multiple POMs, and trying to store both in the same file was causing conflicts.

If you'd still like to have the POMs committed I can create a task that will copy them, and give them sensible names?

Reply:
We'll have to see if Snyk and SonarCloud can find them.

Reply:
Ahh that's the reason they're committed to the repo. I assumed you just wanted to track any changes in the POM!

[Synk can understand Gradle lock files](https://docs.snyk.io/scan-application-code/snyk-open-source/snyk-open-source-supported-languages-and-package-managers/snyk-for-java-and-kotlin#git-services-for-gradle-projects) - would that achieve the same result as committing the POM?

Reply:
@aSamy Yeah, it was just to fool it to think it was a maven project. If the Gradle lock files work, that's fine by me.

Reply:
@ethauvin Great, let's do lock files then.

As far as Sonar support goes, I've had a dig through the Sonar Gradle plugin and the Sonar analyzer for Kotlin https://github.com/SonarSource/sonar-kotlin/tree/master/sonar-kotlin-plugin and from what I can tell Sonar just won't work for Kotlin Multiplatform. The plugin _might_ be hackable to pick up the source files, but the analyzer assumes that the project is structured as a Kotlin/JVM project, and that's difficult to fix.

I can't recommend any alternatives, unfortunately. What do you want to do? Try and find an alternative, disable Sonarqube, or keep urlencoder as JVM only??

Reply:
@aSemy I realized that I never answered you, sorry. We can ditch the Sonar plugin for now. I'm pretty sure I can get Sonarcloud to work without the plugin.

Reply:
Sure thing @ethauvin! I've updated the project, and made some other adjustments.

Reply:
All `gradle.lockfiles` can be updated by running:

```
./gradlew resolveAndLockAllDependencies --write-locks
```